projects/vdk-core/src/vdk/internal/builtin_plugins/connection/managed_cursor.py

```diff
@@ -29,16 +45,41 @@ def __init__(
         self,
         cursor: Any,
         log: logging.Logger = None,
-        connection_hook_spec: ConnectionHookSpec = None,
+        plugin_registry: IPluginRegistry = None,
     ):
         if not log:
             log = logging.getLogger(__name__)
         super().__init__(cursor, log)
-        self.__connection_hook_spec = connection_hook_spec
+        self.__plugin_registry = plugin_registry
+        self.__initialize_cursor_hooks()
+
+    def __initialize_cursor_hooks(self):
+        if self.__plugin_registry:
+            self.__connection_hook_spec = cast(
+                ConnectionHookSpec, self.__plugin_registry.hook()
+            )
+
+            if not self.__plugin_registry.has_plugin(
+                DefaultConnectionHookImpl.__name__
+            ):
+                self.__plugin_registry.load_plugin_with_hooks_impl(
+                    DefaultConnectionHookImpl(), DefaultConnectionHookImpl.__name__
```

Comment:
I moved it 3 times. 

Initially I initialized it in a plugin but that means ManagedCursor does not stand on its own (it's a dependency because the default implementation is necessary for it to work). 

Then I moved it ManagedConnectionBase. But I felt there also it's not a good place for similar reasons. 

I could have created ManagedCursorFactory and initialized it there and used the factory to created new managed cursors but I thought that's too much changes. And also was not sure if it's really good fit. It's the best option perhaps if not this. 

I prefer to merge it this way now and think of refactoring it later when we have better idea on what would be best.

Reply:
Ok. I fixed it by introducing ConnectionHookSpecFactory which is responsible for providing properly initialized ConnectionHookSpec .